### PR TITLE
fix: Fix strain energy calculation for enstatite

### PR DESCRIPTION
**Pull request by adigitoleo**
Don't raise a possibly negative number to a power

### CODE REVIEW

**Comment by adigitoleo** [comment]
slight change from original fortran because I don't understand why line 1437 is necessary if tau_ens is always 1

File: src/pydrex/core.py [hunk 1]
```diff
@@ -628,16 +626,16 @@ def _get_strain_energy(
                 -nucleation_efficiency * dislocation_density ** 2
             )
     elif np.all(rrss == _fabric.RRSS_ENSTATITE):
-        # TODO: What is this? Why is it required?
-        weight_factor = slip_rate_softest * rrss[slip_indices[0]] ** (-stress_exponent)
```
Comment (Patol75) [inline src/pydrex/core.py:632]: Wait, isn't it just `rrss[slip_indices[-1]]`?

File: src/pydrex/core.py [hunk 2]
```diff
@@ -628,16 +626,16 @@ def _get_strain_energy(
...
         # TODO: Verify rrss[i] == τ_0 / τ^sv
         assert rrss[slip_indices[-1]] == 1
-        dislocation_density = rrss[slip_indices[-1]] ** (
+        dislocation_density = 1 ** (
             1.5 - stress_exponent
-        ) * np.abs(weight_factor) ** (1.5 / stress_exponent)
+        ) * np.abs(slip_rate_softest) ** (1.5 / stress_exponent)
```
Comment (Patol75) [inline src/pydrex/core.py:633]: Potentially not needed based on the above.

**Review by Patol75 — CHANGES_REQUESTED** [review]

File: src/pydrex/core.py (hunk 1 above)
Comment (adigitoleo) [reply src/pydrex/core.py:632]: Yeah, that was the first mistake but then I realised that I haven't seen anyone use a value other than 1 for the rrss on the only active slip system of enstatite. However, I suppose it makes more sense to use `slip_indices` if we're already doing that for olivine.

Comment (adigitoleo) [reply src/pydrex/core.py:632]: Based on the fortran I think it is `1 / rrss[slip_indices[-1]]` so that's what I've done in the ammended commit

Comment (Patol75) [reply src/pydrex/core.py:632]: Careful, it is either `(1 / rrss[slip_indices[-1]]) ** stress_exponent` or `rrss[slip_indices[-1] ** -stress_exponent`

**Review by Patol75 — CHANGES_REQUESTED** [review]

File: src/pydrex/core.py [hunk 3]
```diff
@@ -628,16 +626,15 @@ def _get_strain_energy(
                 -nucleation_efficiency * dislocation_density ** 2
             )
     elif np.all(rrss == _fabric.RRSS_ENSTATITE):
-        # TODO: What is this? Why is it required?
-        weight_factor = slip_rate_softest * rrss[slip_indices[0]] ** (-stress_exponent)
-        # TODO: Verify rrss[i] == τ_0 / τ^sv
-        assert rrss[slip_indices[-1]] == 1
+        weight_factor = slip_rate_softest * (1 / rrss[slip_indices[-1]]) ** (-stress_exponent)
```
Comment (Patol75) [inline src/pydrex/core.py:629]: `weight_factor = slip_rate_softest / rrss[slip_indices[-1]] ** stress_exponent`

**Review by Patol75 — CHANGES_REQUESTED** [review]

**Review by Patol75 — APPROVED** [review]

**Pull request merged**